src/ui/components/shared/LoadingScreen.tsx

```diff
@@ -97,31 +97,45 @@ function LoadingScreen({
   uploading,
   awaitingSourcemaps,
   fallbackMessage,
+  isProcessed,
   stalledTimeout = 15000,
-}: PropsFromRedux & { fallbackMessage: string; stalledTimeout?: number }) {
+}: PropsFromRedux & { fallbackMessage: string; isProcessed?: boolean; stalledTimeout?: number }) {
   const longWaitMessage =
     "<div><p>This is taking longer than expected.</p><p><a href='http://replay.io/discord' target='discord'>Contact us on Discord</a></p>";
 
+  const processingMessage = "Processing...";
+
   const [message, setMessage] = useState(fallbackMessage);
+  const [processing, setProcessing] = useState(isProcessed);
```

Comment:
Why not immediately show the processing message? (Why wait for an effect to be run?)
```suggestion
  const [message, setMessage] = useState(isProcessed ? fallbackMessage : "Processing...");
  const [processing, setProcessing] = useState(isProcessed);
```

Seems like it would be better for us to avoid the flicker.